src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/HaivisionXDecoderCommunicator.java

```diff
@@ -692,31 +870,1955 @@ public boolean validatePortRange(Integer portNumber) {
+	 * <li>SrtToUdpTtl</li>
+	 * <li>Encrypted</li>
+	 * <li>Passphrase</li>
+	 * <li>Reject unencrypted caller</li>
+	 * <li>RTSP URL</li>
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param advancedControllableProperties is the list that store all controllable properties
+	 * @param controllableProperty name of controllable property
+	 * @param value value of controllable property
+	 *
+	 * @throws ResourceNotReachableException when fail to create stream
+	 */
+	private void createStreamControl(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties, String streamControllingGroup,
+			String controllableProperty, String value) {
+		StreamControllingMetric streamControllingMetric = StreamControllingMetric.getByName(controllableProperty);
+
+		isEmergencyDelivery = true;
+		switch (streamControllingMetric) {
+			case STREAM_NAME:
+				createStream.setName(value);
+
+				addAdvanceControlProperties(advancedControllableProperties,
+						createText(stats, streamControllingGroup + StreamControllingMetric.STREAM_NAME.getName(), createStream.getName()));
+				populateCancelButtonForCreateStream(stats, advancedControllableProperties);
+				break;
+			case ENCAPSULATION:
+				Encapsulation encapsulationEnum = Encapsulation.getByUiName(value);
+				removeUnusedStatsAndControlByProtocol(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				createStream.setEncapsulation(encapsulationEnum.getApiName());
+
+				populateCreateStreamControl(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				break;
+			case NETWORK_TYPE:
+				NetworkType networkType = NetworkType.getByUiName(value);
+				removeUnusedStatsAndControlByNetworkType(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				switch (networkType) {
+					case UNI_CAST:
+						createStream.setDestinationAddress(DecoderConstant.ADDRESS_ANY);
+						createStream.setSourceAddress(DecoderConstant.ADDRESS_ANY);
+						break;
+					case MULTI_CAST:
+						createStream.setDestinationAddress(DecoderConstant.EMPTY);
+						createStream.setSourceAddress(DecoderConstant.EMPTY);
+						break;
+					default:
+						if (logger.isWarnEnabled()) {
+							logger.warn(String.format("SRT mode %s is not supported.", networkType.getUiName()));
+						}
+						break;
+				}
+
+				populateCreateStreamControl(stats, advancedControllableProperties, createStream, streamControllingGroup);
+				break;
+			case PORT:
+				String port = DecoderConstant.EMPTY;
+				try {
+					Integer portIntValue = Integer.parseInt(value);
+					if (portIntValue < DecoderConstant.MIN_PORT) {
+						portIntValue = DecoderConstant.MIN_PORT;
+					}
+					if (portIntValue > DecoderConstant.MAX_PORT) {
+						portIntValue = DecoderConstant.MAX_PORT;
+					}
+					port = portIntValue.toString();
+				} catch (Exception e) {
+					if (logger.isWarnEnabled()) {
+						logger.warn("Invalid port value", e);
```

Comment:
Are there any other details that could be added to the error message? Something that would help with troubleshooting (port number, stream id or something like that)

Reply:
I fixed it. the new message contains Info to help the user with troubleshooting, 
eg:  "Invalid port value, the port value must be Integer value range to 1025-65535"